includes/functions.php

```diff
@@ -0,0 +1,415 @@
+			'description'       => __( 'The name of your type, at the singular form.', 'bp-types-ui' ),
+			'type'              => 'string',
+			'single'            => true,
+			'sanitize_callback' => 'sanitize_text_field',
+		),
+		'bp_type_has_directory' => array(
+			'label'             => __( 'Add Type-Filtered Directory View', 'bp-types-ui' ),
+			'description'       => __( 'Add a list of members matching the member type available on the Members Directory page (e.g. site.url/members/type/teacher/).', 'bp-types-ui' ),
+			'type'              => 'boolean',
+			'single'            => true,
+			'sanitize_callback' => 'absint',
+		),
+		'bp_type_directory_slug' => array(
+			'label'             => __( 'Custom type directory slug', 'bp-types-ui' ),
+			'description'       => __( 'If you want to use a slug that is different from the Member Type ID above, enter it here.', 'bp-types-ui' ),
+			'type'              => 'string',
+			'single'            => true,
+			'sanitize_callback' => 'sanitize_title',
+		),
+	);
+
+	if ( 'bp_group_type' === $type ) {
+		$meta_args['bp_type_has_directory']['description'] = __( 'Add a list of groups matching the member type available on the Groups Directory page (e.g. site.url/groups/type/ninja/).', 'bp-types-ui' );
+		$meta_args['bp_type_directory_slug']['description'] = __( 'If you want to use a slug that is different from the Group Type ID above, enter it here.', 'bp-types-ui' );
+
+		$meta_args = array_merge(
+			$meta_args,
+			array(
+				'bp_type_show_in_create_screen' => array(
+					'label'             => __( 'Add to Available Types on Create Screen', 'bp-types-ui' ),
+					'description'       => __( 'Include this group type during group creation and when a group administrator is on the group&rsquo;s &ldquo;Manage > Settings&rdquo; page.', 'bp-types-ui' ),
+					'type'              => 'boolean',
+					'single'            => true,
+					'sanitize_callback' => 'absint',
+				),
+				'bp_type_show_in_list'          => array(
+					'label'             => __( 'Include when Group Types are Listed for a Group', 'bp-types-ui' ),
+					'description'       => __( 'Include this group type when group types are listed, like in the group header.', 'bp-types-ui' ),
+					'type'              => 'boolean',
+					'single'            => true,
+					'sanitize_callback' => 'absint',
+				),
+			)
+		);
+	}
+
+	return $meta_args;
+}
+
+/**
+ * Reset BuddyPress type taxonomies to use new arguments.
+ *
+ * @since 1.1.0
+ */
+function bptui_reset_bp_type_taxonomies() {
+	// Reset only once.
+	remove_action( 'bp_register_taxonomies', 'bptui_reset_bp_type_taxonomies', 11 );
+
+	$bp_type_taxonomies = array_intersect(
+		get_taxonomies( array( 'public' => false ) ),
+		array( bp_get_member_type_tax_name(), 'bp_group_type' )
+	);
+
+	$objects = array(
+		bp_get_member_type_tax_name() => 'user',
+		'bp_group_type'               => 'bp_group',
+	);
+
+	foreach ( $bp_type_taxonomies as $bp_type_taxonomy ) {
```

Comment:
Is this necessary just to enable 'show_ui'?

Reply:
Actually: no 😄 . We don't need to unregister/reregister taxonomies. Your question made me look deeper into WordPress and the `register_taxonomy_args` should be used to extend the BP Type taxonomy arguments. That's what I've done in https://github.com/buddypress/bp-types-ui/pull/6/commits/798f2af6960730297ada75918261182e43c3388f

Reply:
Nice!